library/bignum_mod_raw.c

```diff
@@ -176,6 +176,48 @@ void mbedtls_mpi_mod_raw_add( mbedtls_mpi_uint *X,
 
 /* BEGIN MERGE SLOT 6 */
 
+int mbedtls_mpi_mod_raw_canonical_to_modulus_rep(
```

Comment:
I am not certain as weather those two methods belong here. In effect they are performing input validation before calling `mbedtls_mpi_mod_raw_to_mont_rep()`.

Since they will be consumed on the high level api `bignum_mod` they should reside there, as a high level function of the `mbedtls_mpi_mod_raw_to/from_mont_rep`, eg. `mbedtls_mpi_mod_from/to_mont_rep()`

Reply:
These methods work on raw arrays, not on residue structures, so they belong in `mod_raw`.

I'm not sure whether there should be corresponding `mod` structures: when you have a residue, it's supposed to be in the modulus's representation — why would you convert a residue structure? In any case, I didn't need new mod-level conversion functions so I didn't write any.